Python/errors.c

```diff
@@ -306,18 +313,17 @@ PyErr_NormalizeException(PyObject **exc, PyObject **val, PyObject **tb)
     }
     /* normalize recursively */
     tstate = PyThreadState_GET();
-    if (++tstate->recursion_depth > Py_GetRecursionLimit()) {
-        --tstate->recursion_depth;
+    if (++recursion_depth > Py_GetRecursionLimit() - tstate->recursion_depth) {
```

Comment:
Since the function isn't recursive anymore, is there any reason you're still bothering with `recursion_depth`?